polybar-scripts/pulseaudio-microphone/pulseaudio-microphone.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+status() {
+  MUTED=$(pacmd list-sources | awk '/*/,EOF {print}' | awk '/muted/ {print $2; exit}')
+
+  if [ "$MUTED" = "yes" ]; then
+    echo "#1"
+  else
+    echo "#2"
+  fi
+}
+
+listen() {
+    status
+
+    LANG=EN; pactl subscribe | while read -r event; do
+        if echo "$event" | grep -q "source" || echo "$event" | grep -q "server"; then
+            status
+        fi
+    done
+}
+
+toggle() {
+  MUTED=$(pacmd list-sources | awk '/*/,EOF {print}' | awk '/muted/ {print $2; exit}')
+  DEFAULT_SOURCE=$(pacmd list-sources | awk '/*/,EOF {print $3; exit}')
```

Comment:
This line does not work here. Di you try to get the id? That would be the best.

Reply:
Yes, it gets the id of the default source so it can use it to toggle the state. I think the problem was in the regex pattern not working in some distros because of the asterisk. It should be fixed now.